builder/window_builder.sh

```diff
@@ -109,18 +116,18 @@ build_window_icon() {
   if [ "$window_status_icon_enable" = "yes" ]; then
     # #!~[*-]MZ
     local show_window_status=""
-    show_window_status+="#{?window_activity_flag, ${custom_icon_window_activity},}"
-    show_window_status+="#{?window_bell_flag, ${custom_icon_window_bell},}"
-    show_window_status+="#{?window_silence_flag, ${custom_icon_window_silent},}"
-    show_window_status+="#{?window_active, ${custom_icon_window_current},}"
-    show_window_status+="#{?window_last_flag, ${custom_icon_window_last},}"
-    show_window_status+="#{?window_marked_flag, ${custom_icon_window_mark},}"
-    show_window_status+="#{?window_zoomed_flag, ${custom_icon_window_zoom},}"
+    show_window_status+="#{?window_activity_flag,$(prepend_separator "${custom_icon_window_activity}"),}"
+    show_window_status+="#{?window_bell_flag,$(prepend_separator "${custom_icon_window_bell}"),}"
+    show_window_status+="#{?window_silence_flag,$(prepend_separator "${custom_icon_window_silent}"),}"
+    show_window_status+="#{?window_active,$(prepend_separator "${custom_icon_window_current}"),}"
+    show_window_status+="#{?window_last_flag,$(prepend_separator "${custom_icon_window_last}"),}"
+    show_window_status+="#{?window_marked_flag,$(prepend_separator "${custom_icon_window_mark}"),}"
+    show_window_status+="#{?window_zoomed_flag,$(prepend_separator "${custom_icon_window_zoom}"),}"
 
   fi
 
   if [ "$window_status_icon_enable" = "no" ]; then
-    local show_window_status=" #F"
+    local show_window_status=$(prepend_separator "#F")
```

Comment:
This _always_ needs to have a space.

```suggestion
    local show_window_status=" #F"
```

Reply:
> This _always_ needs to have a space.

The reason I changed it here is because I wanted consistency of code between custom icons and text display of window status. And it is going to **always** have space, because tmux patterns are not yet expanded at that point, so it is just "#F" which is not empty and hence the space separator is always prepended.

Reply:
> The reason I changed it here is because I wanted consistency of code

Consistency is not adding dead/useless code, especially not dead/useless code that makes it _harder_ to read.

Reply:
Restored